### PR TITLE
feat(utils): add list,dict utils

### PR DESCRIPTION
**Pull request by tepelbaum**
**Type of change**
- [x] Feature
- [ ] Bugfix
- [ ] Refactoring
- [ ] Documentation

-----------------------

**Context**

>_What does this PR implement and how? What are the requirements?_

>_Why is this change needed? Related issue #?_

>_Any diagrams or screenshot to help reviewers?_

>_Please describe the tests you have performed to ensure the feature or fix are robust & effective?_

Your comments

-----------------------

**What should the reviewer focus on?**
>_Do you have specific questions/ areas for the reviewer?_

>_Which alternative solutions have you already considered and why did you not implement it?_

>_Where is the highest risk/ most complicated change that the reviewer should focus on?_

>_Is there a specific code review topic the reviewer should focus on (e.g., correct error handling, API usage)_

>_What is a good place to start the review (e.g., specific file/method?)_

Your comments

-----------------------

**For reviewer:**

[PR Review guidelines](https://google.github.io/eng-practices/review/reviewer/standard.html)

### CODE REVIEW

File: ecodev_core/list_utils.py [hunk 1]
```diff
@@ -45,6 +45,45 @@ def first_or_default(sequence: Union[List[Any], None],
     return next((elt for elt in sequence if condition(elt)), default)
 
 
+def sort_by_keys(unsorted_dict: dict, reverse: bool = False) -> dict:
```
Comment (jyates0eco) [inline ecodev_core/list_utils.py:48]: Curious as to the use case here as I don't think I've ever seen a need for a sorted dictionary?

File: ecodev_core/list_utils.py [hunk 2]
```diff
@@ -45,6 +45,45 @@ def first_or_default(sequence: Union[List[Any], None],
...
+    return dict(sorted(unsorted_dict.items(), key=lambda item: item[1], reverse=reverse))
+
+
+def first_func_or_default(sequence: list[Callable] | None,
+                          elt: Any,
+                          condition: Callable | None = None,
+                          default: Any | None = None
+                          ) -> Any | None:
+    """
+    Returns the first element of a functional sequence is a certain criteria is met
```
Comment (jyates0eco) [inline ecodev_core/list_utils.py:72]:
```suggestion
    Returns the first element of a functional sequence if a certain criteria is met
```

File: ecodev_core/list_utils.py (hunk 1 above)
Comment (tepelbaum) [reply ecodev_core/list_utils.py:48]: From keys it's rarer, but it happens from time to time :)

Comment (tepelbaum) [reply ecodev_core/list_utils.py:48]: I have been 54 times on https://stackoverflow.com/questions/9001509/how-do-i-sort-a-dictionary-by-key ahahah

File: tests/unitary/test_list_utils.py [hunk 3]
```diff
@@ -55,3 +58,25 @@ def test_first_transformed_or_default(self):
         """
         data = [1, 2, 3, 1, 1, 2, 3]
         self.assertEqual(first_transformed_or_default(data, lambda x: x * 3 if x > 2 else None), 9)
+
+    def test_first_func_or_default(self):
```
Comment (jyates0eco) [inline tests/unitary/test_list_utils.py:62]: ❤️

**Review by jyates0eco — APPROVED** [review]
Lovely!!!! Super cool to see the dict reversals 😍

File: ecodev_core/list_utils.py (hunk 1 above)
Comment (jyates0eco) [reply ecodev_core/list_utils.py:48]: it's a very popular one it seems! I dunno what rock I've been hiding under hahaha

**Pull request merged**